File: JavaScript/7js 1/countWins.js
/*
|--------------------------------------------------------------------------
| Find how many times did a team from a given country win the Champions League?
|--------------------------------------------------------------------------
|
*/

function countWins(winnerList, country) {
   return winnerList.filter(i => i.country == country).length
}


function countWins(winnerList, country) {
   return winnerList.reduce((pre, val) => pre + (val.country === country), 0);
}


function countWins(winnerList, country) {
   let count = 0;
   winnerList.forEach(elem => {
      if (elem.country === country) count++;
   })
   return count;
}


console.log(countWins(winnerList1, 'Portugal'), 1);
console.log(countWins(winnerList1, 'FootLand'), 0);
console.log(countWins(winnerList1, 'Spain'), 9);

File: Python/7py 1/countWins.py
"""
|--------------------------------------------------------------------------
| Find how many times did a team from a given country win the Champions League?
|--------------------------------------------------------------------------
|
"""

def countWins(winnerList, country):
   return len(list(filter(lambda i: i['country'] == country, winnerList)))


def countWins(winners, country):
   return str(winners).count(country)


def countWins(winnerList, country):
   return sum(x.get('country') == country for x in winnerList)


def countWins(winnerList, country):
   return sum(i["country"] == country for i in winnerList)